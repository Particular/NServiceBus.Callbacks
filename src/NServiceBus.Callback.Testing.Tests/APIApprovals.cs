using NServiceBus.Callbacks.Testing;
using NUnit.Framework;
using Particular.Approvals;
using PublicApiGenerator;

[TestFixture]
public class APIApprovals
{
    [Test]
    public void Approve()
    {
        var publicApi = ApiGenerator.GeneratePublicApi(typeof(TestableCallbackAwareSession).Assembly);
        Approver.Verify(publicApi);
    }
}