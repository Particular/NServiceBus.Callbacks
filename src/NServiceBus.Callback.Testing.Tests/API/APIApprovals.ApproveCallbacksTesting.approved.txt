[assembly: System.CLSCompliantAttribute(true)]
[assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]
namespace NServiceBus.Callbacks.Testing
{
    public class TestableCallbackAwareSession : NServiceBus.Testing.TestableMessageSession
    {
        public TestableCallbackAwareSession() { }
        public override async System.Threading.Tasks.Task Send(object message, NServiceBus.SendOptions options) { }
        public void When<TRequest, TResult>(System.Func<TRequest, bool> matcher, TResult response)
            where TRequest :  class { }
        public void When<TRequest, TResult>(System.Func<TRequest, NServiceBus.SendOptions, bool> matcher, TResult response)
            where TRequest :  class { }
    }
}